### PR TITLE
Feature/PN-31 POC sync of messages in cluster

### PR DESCRIPTION
**Pull request by cosminrentea**

### CODE REVIEW

File: server/cluster/codec.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+type messageType int
+
+var (
+	mh codec.MsgpackHandle
+	h  = &mh // or mh to use msgpack
+)
+
+const (
+	NEXT_ID_RESPONSE messageType = iota
```
Comment (Bogh) [inline server/cluster/codec.go]: CamelCase constants please.

File: gubled/config/config.go [hunk 2]
```diff
@@ -73,6 +74,24 @@ var (
 			Envar("GUBLE_GCM_WORKERS").
 			Int(),
 	}
+	// Cluster settings related to activating the cluster module
+	Cluster = struct {
+		NodeID   *int
+		NodePort *int
+		Remotes  *[]*url.URL
+	}{
+		NodeID: kingpin.Flag("node-id", "This guble node's own ID (used in cluster mode): a strictly positive integer number which must be unique in cluster").
```
Comment (Bogh) [inline gubled/config/config.go]: Make sure the value is bigger than 0.

Comment (cosminrentea) [inline gubled/config/config.go]: It can be zero by default -> standalone.
Only when starting as cluster, it is validated to be >0

Comment (cosminrentea) [inline gubled/config/config.go]: however, to eliminate any misunderstanding, I will validate it right from the start

**Pull request merged**